lib/gooddata/lcm/actions/collect_data_product.rb

```diff
@@ -0,0 +1,62 @@
+# encoding: UTF-8
+#
+# Copyright (c) 2010-2017 GoodData Corporation. All rights reserved.
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+require_relative 'base_action'
+
+module GoodData
+  module LCM2
+    class CollectDataProduct < BaseAction
+      DESCRIPTION = 'Collect DataProduct'
+
+      PARAMS = define_params(self) do
+        description 'Client Used for Connecting to GD'
+        param :gdc_gd_client, instance_of(Type::GdClientType), required: true
+
+        description 'DataProduct to manage'
+        param :data_product, instance_of(Type::StringType), required: false
+      end
+
+      RESULT_HEADER = [
+        :data_product
+      ]
+
+      class << self
+        def call(params)
+          params = params.to_hash
+          client = params.gdc_gd_client
+          domain_name = params.organization || params.domain
+          domain = client.domain(domain_name) || fail("Invalid domain name specified - #{domain_name}")
+
+          if params.key?(:data_product)
+            begin
+              data_product = domain.data_products(params.data_product)
+            rescue RestClient::BadRequest
+            #   this is handled correctly in ensure_data_product.rb
```

Comment:
Style/CommentIndentation: Incorrect indentation detected (column 12 instead of 14).